src/modules/workflows/workflowEngine.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable indent */
+const { Status, WorkflowKeysEnum } = require('./enums');
+
+module.exports = (workflow, { writer, editor }) => {
+  const currentStatus = workflow.status;
+  const changeLog = {};
+
+  switch (currentStatus) {
+    case Status.WAITING_FOR_WRITER:
+      if (!writer) {
+        throw new Error('A writer must be provided to proceed to the next step.');
+      }
+      changeLog[WorkflowKeysEnum.writer] = writer;
+      changeLog[WorkflowKeysEnum.status] = Status.ASSIGNED_TO_WRITER;
+      break;
+
+    case Status.ASSIGNED_TO_WRITER:
+      if (writer) {
+        changeLog[WorkflowKeysEnum.writer] = writer;
+      } else {
+        changeLog[WorkflowKeysEnum.status] = Status.WRITING_IN_PROGRESS;
+      }
+      break;
+
+    case Status.WRITING_IN_PROGRESS:
```

Comment:
how to handle save for later?

asignedToWriter ---> saveforlater -> next
writinginpreogress-- > saveforlater -> auditlog

asignedToWriter ---> saveforlater -> next
writinginpreogress-- > saveforlater (wip) -> next